File: Questions/Longest Increasing Subsequence/Answer.cpp
int Solution::lis(const vector<int> &A) {
    if (A.empty()) {
        return 0;
    }

    int n = A.size();
    vector<int> dp(n, 1);

    for (int i = 1; i < n; ++i) {
        for (int j = 0; j < i; ++j) {
            if (A[i] > A[j] && dp[i] < dp[j] + 1) {
                dp[i] = dp[j] + 1;
            }
        }
    }

    return *max_element(dp.begin(), dp.end());
}

File: Questions/Stairs/Answer.java
public class Solution {
    public int climbStairs(int A) {  
        if (A == 1) {
            return 1;
        }
        if (A == 2) {
            return 2;
        }
        int[] dp = new int[A + 1];
        dp[1] = 1;
        dp[2] = 2;
        for (int i = 3; i <= A; i++) {
            dp[i] = dp[i - 1] + dp[i - 2];
        }
        return dp[A];
    }
    public static void main(String[] args) {
        Solution solution = new Solution();
        System.out.println(solution.climbStairs(2)); 
        System.out.println(solution.climbStairs(3)); 
    }
}
